operator/controller/src/main/java/io/apicurio/registry/operator/resource/ResourceFactory.java

```diff
@@ -254,6 +260,19 @@ private void addSelectorLabels(Map<String, String> labels, ApicurioRegistry3 pri
         // spotless:on
     }
 
+    private int getRegistryBackendReplicas(ApicurioRegistry3SpecApp registry3SpecApp) {
+        return registry3SpecApp.getReplicas() != null ? registry3SpecApp.getReplicas() : DEFAULT_REPLICAS;
```

Comment:
I've started using `Optional.ofNullable(primary).map(field..).map(field...).orElse(default);` idiom for traversing the CR, since it's easier to handle `null` somewhere in the nested structure.